cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientFactory.java

```diff
@@ -77,54 +75,52 @@ public CloudFoundryClientFactory getCloudFoundryClientFactory() {
-    }
-
-    public CloudControllerRestClient createClient(URL controllerUrl, CloudCredentials credentials, String organizationName,
-                                                  String spaceName, OAuthClient oAuthClient) {
+                                                  String spaceName, OAuthClient oAuthClient, List<ExchangeFilterFunction> exchangeFilters) {
         CloudControllerRestClient clientWithoutTarget = createClient(controllerUrl, credentials, oAuthClient);
         CloudSpace target = clientWithoutTarget.getSpace(organizationName, spaceName);
 
-        return createClient(controllerUrl, credentials, target, oAuthClient);
+        return createClient(controllerUrl, credentials, target, oAuthClient, exchangeFilters);
     }
 
-    public CloudControllerRestClient createClient(URL controllerUrl, CloudCredentials credentials, CloudSpace target) {
-        return createClient(controllerUrl, credentials, target, createOAuthClient(controllerUrl, credentials.getOrigin()));
+    public CloudControllerRestClient createClient(URL controllerUrl, CloudCredentials credentials, OAuthClient oAuthClient) {
+        return createClient(controllerUrl, credentials, null, oAuthClient, null);
     }
 
-    public CloudControllerRestClient createClient(URL controllerUrl, CloudCredentials credentials, CloudSpace target,
-                                                  OAuthClient oAuthClient) {
-        RestTemplate restTemplate = createAuthorizationSettingRestTemplate(credentials, oAuthClient);
-        CloudFoundryClient delegate = getCloudFoundryClientFactory().createClient(controllerUrl, oAuthClient);
-        DopplerClient dopplerClient = getCloudFoundryClientFactory().createDopplerClient(controllerUrl, oAuthClient);
-
-        return new CloudControllerRestClientImpl(controllerUrl, credentials, restTemplate, oAuthClient, delegate, dopplerClient, target);
+    public CloudControllerRestClient createClient(URL controllerUrl, CloudCredentials credentials, CloudSpace target) {
+        return createClient(controllerUrl, credentials, target, createOAuthClient(controllerUrl, credentials.getOrigin()), null);
     }
 
     private OAuthClient createOAuthClient(URL controllerUrl, String origin) {
         Map<String, Object> infoMap = getInfoMap(controllerUrl);
         URL authorizationEndpoint = getAuthorizationEndpoint(infoMap);
         if (StringUtils.isEmpty(origin)) {
-            return restUtil.createOAuthClient(authorizationEndpoint, getHttpProxyConfiguration(), shouldTrustSelfSignedCertificates());
+            return restUtil.createOAuthClient(authorizationEndpoint, shouldTrustSelfSignedCertificates());
         }
         ConnectionContext connectionContext = getCloudFoundryClientFactory().getOrCreateConnectionContext(controllerUrl.getHost());
-        return restUtil.createOAuthClient(authorizationEndpoint, getHttpProxyConfiguration(), shouldTrustSelfSignedCertificates(),
```

Comment:
What happened with that configuration getHttpProxyConfiguration()

Reply:
This proxy configuration was never used anywhere. It was a remnant from previous changes so I removed it.